model/command_response.go

```diff
@@ -37,6 +41,40 @@ func (o *CommandResponse) ToJson() string {
 	return string(b)
 }
 
+func (o *CommandResponse) IsValid() error {
+	var errs *multierror.Error
+	if o.Text == "" && len(o.Attachments) == 0 {
+		errs = multierror.Append(errs, errors.New("'text' or 'attachments' is required"))
+	}
+	if o.ResponseType != "" && o.ResponseType != "ephemeral" && o.ResponseType != "in_channel" {
+		errs = multierror.Append(errs, errors.New("'response_type' must be empty, 'ephemeral', or 'in_channel'"))
+	}
+	if _, err := url.ParseRequestURI(o.GotoLocation); o.GotoLocation != "" && err != nil {
+		errs = multierror.Append(errs, errors.New("'goto_location' must be a valid URI"))
+	}
+	if o.Type != "" && !strings.HasPrefix(o.Type, "custom_") {
```

Comment:
Actually, I don't think our documentation is correct here. At least I don't reminder anything that enforces a `custom_` prefix.

@lieut-data Do you know? You added this line in https://github.com/mattermost/mattermost-developer-documentation/pull/100.

Reply:
It looks to be enforced here: https://github.com/mattermost/mattermost-server/blob/b30acfdee6fc85bd08a1d3dc6c1dd8b337188310/model/post.go#L319-L321

Reply:
(Note that my documentation and this code is about custom /post/ types. I don't know if it naturally extends to command responses or not, but wouldn't surprise me given they can end up as posts.)